### ISSUE TITLE
returning error response

### ISSUE THREAD

**Issue by copenju** [Mwangangi/ci-africastalking#3]
Hi a good tool just wanted to inquire how to return responses in case the SMS was not send based on your tool am not good in PHP
//Using the default sender ID

$this->africastalking->sendMessage($recipients, $message);

///??response stutus/////

**Comment by Mwangangi** [comment]
@SamuelMwangiW kindly assist

**Comment by SamuelMwangiW** [comment]
@copenju Kindly confirm if you are using Codeigniter or any framework at all.

**Comment by SamuelMwangiW** [comment]
This package was written a while back for Codeigniter 3. If the config's properly configured with AT Username and API, calling `$this->africastalking->sendMessage($recipients, $message);` with an array of recipients and the string message, the method returns the Africastalking response object.
https://github.com/Mwangangi/ci-africastalking/blob/bb2601bcb3e8e5d7e9cc25ae2c4dc23db117e995/application/libraries/Africastalking.php#L19-L52
@Mwangangi in your opinion, should we upgrade the package to CI4?

**Comment by Mwangangi** [comment]
Yes, the project needs an upgrade to CI4. However, I am not able to handle that at the moment. I'm too held up to make such a commitment.  @SamuelMwangiW if you're able to, kindly do.

**Comment by copenju** [comment]
Yes please kindly if you the capacity to upgrade it will be okay I support
this

**Comment by copenju** [comment]
ok hi mwangi this all we to do in the controller, just call this sendMessage() and assign it a variable
//..............
 $result_array=$this->africastalking->sendMessage($recipient["contact"], $message);
//logic here ...

**Comment by SamuelMwangiW** [comment]
@copenju kindly paste here the error you are getting

**Comment by copenju** [comment]
> @copenju kindly paste here the error you are getting

ok I had a problem with getting the response but I have sorted it out as you explained earlier , it's now ok I was tagging @Mwangangi, the names look the same

**Comment by SamuelMwangiW** [comment]
@Mwangangi Its been years now since I was using CodeIgniter. I now see that CI4 supports composer natively out of the box. I will spare some time over a week and create a simple CI library based on composer

**Comment by copenju** [comment]
> @Mwangangi Its been years now since I was using CodeIgniter. I now see that CI4 supports composer natively out of the box. I will spare some time over a week and create a simple CI library based on composer

ok that will be great its also as if Africas talking are no longer advocating PHP library

**Comment by SamuelMwangiW** [comment]
@copenju That's actually not the case, the offcial SDK is actively developed repo and you can find it here https://github.com/AfricasTalkingLtd/africastalking-php and their docs also prominently include PHP code samples (https://developers.africastalking.com/docs/sms/sending/bulk). This was a wrapper around the official SDK

**Comment by copenju** [comment]
Thanks, I'll check them out.

**Comment by GurupriyaWebvillee** [comment]
Hello,
i am getting this error 

An uncaught Exception was encountered
Type: AfricasTalkingGatewayException

Message: The supplied authentication is invalid

Filename: /var/www/html/sery_new/application/third_party/AfricasTalkingGateway.php

Line Number: 103

Backtrace:

File: /var/www/html/sery_new/application/controllers/Admin.php
Line: 3258
Function: sendMessage

File: /var/www/html/sery_new/index.php
Line: 315
Function: require_once

**Comment by GurupriyaWebvillee** [comment]
i am using like this 

public function sms(){
        $this->load->library('africastalking');

        $recipients = "89685748574";
        $message = "fghh";
        $this->africastalking->sendMessage($recipients, $message); 

    }

**Comment by copenju** [comment]
is that phone number correct

**Comment by SamuelMwangiW** [comment]
@GurupriyaWebvillee I also suspect the username and api key supplied in the config might also be a problem given the error:

```
Message: The supplied authentication is invalid
```

**Comment by GurupriyaWebvillee** [comment]
I am giving  username -  sandbox, and  key  - sandbox key. and also i am giving now my correct phone number.

**Controller -** 
public function sms(){
        $this->load->library('africastalking');

        $recipient["contact"] = "8770186120";
        $message = "Test message";
        $sender_id = 'TestSender';
        $result_array = $this->africastalking->sendMessage($recipients, $message,$sender_id);

    }

**Config-**
$config = array(
    'username'                  => 'sandbox',
    'apiKey'                    => '1bdff728b5ddc1e9638d8a98c07c0be699e431f8c8538a91341e19b3d734114c',
    'default_country_code'      => '+91',//254
    'sms_sender'                => 'AFRICASTKNG'            //Leave as NULL to send using the default senderId 'AFRICASTKNG'
    );

**Error**
An uncaught Exception was encountered
Type: AfricasTalkingGatewayException

Message: Please supply both to and message parameters

Filename: /var/www/html/sery_new/application/third_party/AfricasTalkingGateway.php

Line Number: 57

Backtrace:

File: /var/www/html/sery_new/application/controllers/Admin.php
Line: 3259
Function: sendMessage

File: /var/www/html/sery_new/index.php
Line: 315
Function: require_once
